test/test_buffer.cpp

```diff
@@ -1,11 +1,160 @@
+#include <array>
 #include "gtest/gtest.h"
 #include "buffer.h"
 
+
 TEST(CRcvBuffer, Create)
 {
-    const int buffer_size = 65536;
+    const int buffer_size_pkts = 128;
+    CUnitQueue unit_queue;
+    CRcvBuffer rcv_buffer(&unit_queue, buffer_size_pkts);
+
+    EXPECT_EQ(rcv_buffer.getAvailBufSize(), buffer_size_pkts - 1);   // logic
+}
+
+
+TEST(CRcvBuffer, FullBuffer)
+{
+    const int buffer_size_pkts = 16;
+    CUnitQueue unit_queue;
+    unit_queue.init(buffer_size_pkts, 1500, AF_INET);
+    CRcvBuffer rcv_buffer(&unit_queue, buffer_size_pkts);
+
+    const size_t payload_size = 1456;
+    // Add a number of units (packets) to the buffer
+    // equal to the buffer size in packets
+    for (int i = 0; i < rcv_buffer.getAvailBufSize(); ++i)
+    {
+        CUnit* unit = unit_queue.getNextAvailUnit();
+        EXPECT_NE(unit, nullptr);
+        unit->m_Packet.setLength(payload_size);
+        EXPECT_EQ(rcv_buffer.addData(unit, i), 0);
```

Comment:
I think as you are using `getAvailBufSize`, which isn't a straightforward forwarder to a field, you might also do a check if the number of iterations in this loop has counted to `buffer_size_pkts`. Just a proposal.